node/src/reactor/joiner.rs

```diff
@@ -407,6 +425,9 @@ impl reactor::Reactor for Reactor {
 
         let metrics = Metrics::new(registry.clone());
 
+        let (console, console_effects) =
+            Console::new(&WithDir::new(&root, config.console.clone()), event_queue)?;
```

Comment:
Do we need to create an instance of it if we can't handle the requests? Can't we just _not_ create it?

Reply:
Oh, I think I see - we should be accepting new connections but we won't handle the command to dump consensus' state.

Reply:
And just for the sake of completion, when reactor transitions from `joining` to `participating`, it will drop the socket and print this to the console:
```
2022/01/17 01:18:00 socat[1881] E write(5, 0x55e4ef777f10, 8): Broken pipe
```

Reply:
> Oh, I think I see - we should be accepting new connections but we won't handle the command to dump consensus' state.

Yes! We could just not create it of course, but not being able to handle any requests is temporary state of the feature - soon there should be requests that can be answered in the joiner as well: #2546. So while we could omit it for now, I think it's better to add it here straight away.

> But won't we drop all connections when transitioning from joiner to participating reactor? Since the sender_shutdown gets dropped.

Yes. This is a bit of a wrinkle, but I think I can live with that for now. It's the same with any other component that gets re-created.

> And just for the sake of completion, when reactor transitions from joining to participating, it will drop the socket and print this to the console:

That's just socat reporting the connection has been closed.